app/src/main/kotlin/at/bitfire/davdroid/ui/AccountsActivity.kt

```diff
@@ -6,53 +6,17 @@ package at.bitfire.davdroid.ui
 
 import android.content.Intent
 import android.os.Bundle
-import androidx.activity.compose.setContent
 import androidx.appcompat.app.AppCompatActivity
-import at.bitfire.davdroid.ui.account.AccountActivity
-import at.bitfire.davdroid.ui.intro.IntroActivity
-import at.bitfire.davdroid.ui.setup.LoginActivity
-import dagger.hilt.android.AndroidEntryPoint
-import javax.inject.Inject
 
-
-@AndroidEntryPoint
+@Deprecated("Automatically redirects to MainActivity. Should be removed in the future.")
 class AccountsActivity: AppCompatActivity() {
-
-    @Inject
-    lateinit var accountsDrawerHandler: AccountsDrawerHandler
-
-    private val introActivityLauncher = registerForActivityResult(IntroActivity.Contract) { cancelled ->
-        if (cancelled)
-            finish()
-    }
-
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        // handle "Sync all" intent from launcher shortcut
-        val syncAccounts = intent.action == Intent.ACTION_SYNC
-
-        setContent {
-            AccountsScreen(
-                initialSyncAccounts = syncAccounts,
-                onShowAppIntro = {
-                    introActivityLauncher.launch(null)
-                },
-                accountsDrawerHandler = accountsDrawerHandler,
-                onAddAccount = {
-                    startActivity(Intent(this, LoginActivity::class.java))
-                },
-                onShowAccount = { account ->
-                    val intent = Intent(this, AccountActivity::class.java)
-                    intent.putExtra(AccountActivity.EXTRA_ACCOUNT, account)
-                    startActivity(intent)
-                },
-                onManagePermissions = {
-                    startActivity(Intent(this, PermissionsActivity::class.java))
-                }
-            )
-        }
+        startActivity(
+            Intent(this, MainActivity::class.java).apply {
+                action = intent.action
+            }
+        )
```

Comment:
I think we need some "replace current activity" logic. This code just launches a new `MainActivity` on top of the current `AccountsActivity`, but doesn't replace it. Which leads to a strange behavior:

1. Force-close all open apps etc., clear task stack
2. `adb shell am start-activity at.bitfire.davdroid/.ui.AccountsActivity` to simulate a third-party app
3. DAVx5 opens, it will correctly open the accounts screen.
4. However: now press Back. It should just go back.Instead, it stays in a white empty screen (Android 15 emulator). If you press Back again, it goes back to the Accounts screen again.